sentry_sdk/scope.py

```diff
@@ -134,14 +134,30 @@ def fingerprint(self, value):
         """When set this overrides the default fingerprint."""
         self._fingerprint = value
 
-    @_attr_setter
+    @property
+    def transaction(self):
+        # type: () -> Optional[Span]
+        """Return the transaction (root span) in the scope."""
+        # XXX: update return type to Optional[Transaction]
+        try:
+            return self._span._span_recorder.spans[0]
+        except (AttributeError, IndexError):
+            return None
+
+    @transaction.setter
     def transaction(self, value):
         # type: (Optional[str]) -> None
         """When set this forces a specific transaction name to be set."""
+        # XXX: the docstring above is misleading. The implementation of
+        # apply_to_event prefers an existing value of event.transaction over
+        # anything set in the scope.
+        # XXX: note that with the introduction of the Scope.transaction getter,
+        # there is a semantic and type mismatch between getter and setter. The
+        # getter returns a transaction, the setter sets a transaction name.
+        # Without breaking version compatibility, we could make the setter set a
+        # transaction name or transaction (self._span) depending on the type of
+        # the value argument.
         self._transaction = value
-        span = self._span
-        if span:
-            span.transaction = value
```

Comment:
This is already done in `Scope.apply_to_event`. Removing from here to match the behavior with JS.

Note that setting `span.transaction` here may inadvertently promote a regular span into a transaction; we're leaning towards a clearer conceptual separation between transactions and spans. The separation should become more clear with https://github.com/getsentry/sentry-python/pull/715, that introduces the `Transaction` class and makes `Span.transaction` => `Transaction.name`.

Reply:
This is not done in apply_to_event, only the trace context is being set

Reply:
https://github.com/getsentry/sentry-python/blob/b539ecb9c6a8c990051ccc5d7d0d80f8723f6a3b/sentry_sdk/scope.py#L295-L296

With the difference that in JS there is no equivalent of `event.get("transaction") is None` as the first condition, but I don't want to touch that detail in this PR, as it is changing a lot of how Scope works and is unrelated to Tracing.

Reply:
I give up removing this for now. The test `test_continue_from_headers` implies an expectation that setting `scope.transaction` overwrites the name of the current transaction. In JS changing the scope only changes the scope, and the scope is applied to events/transactions later.